view/frontend/web/js/view/payment/method-renderer/omise-offsite-tesco-method.js

```diff
@@ -0,0 +1,136 @@
+             */
+            getData: function() {
+                return {
+                    'method': this.item.method,
+                    'additional_data': {
+                        'offsite': this.omiseOffsite()
+                    }
+                };
+            },
+
+            /**
+             * Is method available to display
+             *
+             * @return {boolean}
+             */
+            isActive: function() {
+                return true;
+            },
+
+            /**
+             * Hook the placeOrder function.
+             * Original source: placeOrder(data, event); @ module-checkout/view/frontend/web/js/view/payment/default.js
+             *
+             * @return {boolean}
+             */
+            placeOrder: function(data, event) {
+                var self = this;
+
+                if (event) {
+                    event.preventDefault();
+                }
+
+                self.getPlaceOrderDeferredObject()
+                    .fail(
+                        function(response) {
+                            errorProcessor.process(response, self.messageContainer);
+                            fullScreenLoader.stopLoader();
+                            self.isPlaceOrderActionAllowed(true);
+                        }
+                    ).done(
+                        function(response) {
+                            var self = this;
+
+                            var serviceUrl = urlBuilder.createUrl(
+                                '/orders/:order_id/omise-offsite',
+                                {
+                                    order_id: response
+                                }
+                            );
+
+                            storage.get(serviceUrl, false)
+                                .fail(
+                                    function (response) {
+                                        errorProcessor.process(response, self.messageContainer);
+                                        fullScreenLoader.stopLoader();
+                                        self.isPlaceOrderActionAllowed(true);
+                                    }
+                                )
+                                .done(
+                                    function (response) {
+                                        if (response) {
+                                            redirectOnSuccessAction.execute();
+                                        } else {
+                                            errorProcessor.process(response, self.messageContainer);
+                                            fullScreenLoader.stopLoader();
+                                            self.isPlaceOrderActionAllowed(true);
+                                        }
+                                    }
+                                );
+                        }
+                    );
+
+                return true;
+            }
+        });
+    }
```

Comment:
Lots of repeated, identical or near identical functions here for no apparent reason

Reply:
what functions @jonrandy ? Identical to what other functions?

Reply:
Identical to other functions that are the same as them

Reply:
:)

Reply:
```
                        function(response) {
                            errorProcessor.process(response, self.messageContainer);
                            fullScreenLoader.stopLoader();
                            self.isPlaceOrderActionAllowed(true);
                        }
```
This appears effectively 3 times

Reply:
Would make more sense to split out into 1 function and call it?

Reply:
you mean other payment methods?

Reply:
I mean the function above